bindings/java/src/org/sleuthkit/datamodel/blackboardutils/CommunicationArtifactsHelper.java

```diff
@@ -770,10 +770,30 @@ public void addAttachments(BlackboardArtifact message, MessageAttachments attach
 		// Create attribute 
 		message.addAttribute(new BlackboardAttribute(BlackboardAttribute.ATTRIBUTE_TYPE.TSK_ATTACHMENTS, getModuleName(), attachmentsJson));
 
-		// @TODO 5698: create a TSK_ASSOCIATED_OBJECT artifact for each fileAttachment 
-		//             to associate the file with the message.
+		// Associate each attachment file with the message.
+		Collection<FileAttachment> fileAttachments = attachments.getFileAttachments();
+		for (FileAttachment fileAttachment : fileAttachments) {
+			long attachedFileObjId = fileAttachment.getObjectId();
+			if (attachedFileObjId >= 0) {
+				AbstractFile attachedFile = message.getSleuthkitCase().getAbstractFileById(attachedFileObjId);
+				associateAttachmentWithMesssge(message, attachedFile);
+			}
+		}
 	}
 
+	/**
+	 * Creates a TSK_ASSOCIATED_OBJECT artifact between the attachment file and 
+	 * the message.
+	 */
```

Comment:
Missing param docs.

Reply:
I can certainly add it, but this is a private method, not an API.